messages/eps_fhir/dispense_notification.py

```diff
@@ -15,7 +15,8 @@ class DispenseNotificationIDs:
 class DispenseNotification:
     def __init__(self, context: Any) -> None:
         ids = DispenseNotificationIDs()
-
+        self.context = context
+        self.context.dn_id = str(uuid4())
```

Comment:
Can it be dispense_notification_id please?

And can it be done in the same way as in prescription.py too? So pass the context into the IDs class's __init__ (the class will no longer be a dataclass).